fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/entity/FakePlayer.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.UUID;
+
+import com.google.common.collect.MapMaker;
+import com.mojang.authlib.GameProfile;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.entity.SignBlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.entity.passive.AbstractHorseEntity;
+import net.minecraft.inventory.Inventory;
+import net.minecraft.network.packet.c2s.play.ClientSettingsC2SPacket;
+import net.minecraft.scoreboard.AbstractTeam;
+import net.minecraft.screen.NamedScreenHandlerFactory;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.stat.Stat;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.fabric.impl.event.interaction.FakePlayerNetworkHandler;
+
+/**
+ * A "fake player", i.e. a {@link ServerPlayerEntity} that is not a human player.
+ * They are typically used to automatically perform player actions such as placing blocks.
+ *
+ * <p>The easiest way to obtain a fake player is with {@link FakePlayer#get(ServerWorld, GameProfile)}.
+ * It is also possible to create a subclass for more control over the fake player's behavior.
+ * For good inter-mod compatibility, fake players should have the UUID of their owning (human) player.
+ *
+ * <p>Fake players try to behave like regular {@link ServerPlayerEntity} objects to a reasonable extent.
+ * In some edge cases, or for gameplay considerations, it might be necessary to check whether a {@link ServerPlayerEntity} is a fake player.
+ * This can be done with an {@code instanceof} check: {@code player instanceof FakePlayer}.
+ */
+public class FakePlayer extends ServerPlayerEntity {
+	/**
+	 * Default UUID, for fake players not associated with a specific (human) player.
+	 */
+	public static final UUID DEFAULT_UUID = UUID.fromString("41C82C87-7AfB-4024-BA57-13D2C99CAE77");
```

Comment:
Add a comment explaining why this specific UUID is chosen?

Reply:
```suggestion
	public static final UUID DEFAULT_UUID = UUID.fromString("FAB41CF4-B41C-FAB1-41C1-FAB41CFAB41C");
```

Reply:
I said it's the one used by Forge, but it doesn't really need a comment, could very well have been any one UUID.